### PR TITLE
check if kubectl version required values are empty

### LINKED ISSUE #95297: e2e kubectl version check (and internal API test) should check that all version values are not empty
[The e2e tests for `kubectl version`](https://github.com/kubernetes/kubernetes/blob/16bc6c1d02fc8e4d40ee937ce0fdf7313ce2f233/test/e2e/kubectl/kubectl.go#L1398-L1413) only check that the output includes the correct *keys* - they don't confirm that it has appropriate *values*. Doing full validation on the version data, or at least enough to catch null-string values, would guard against failures like the bad functionality seen in OpenSUSE's binaries, which have been shipping with many of the fields left completely blank (as I discovered [when a line of code reading these values broke provisioning in rook-ceph](https://github.com/rook/rook/issues/6364#issuecomment-703492592)).

### PR DESCRIPTION
**Pull request by SaiHarshaK**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
fix e2e kubectl version test to check that required values are not empty with regex

**Which issue(s) this PR fixes**:
Fixes #95297

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/kubectl/kubectl.go
Comment (dougsland) [inline test/e2e/kubectl/kubectl.go:1514]: I would add one line comment above the regex with some explanation.

File: test/e2e/kubectl/kubectl.go
Comment (dougsland) [inline test/e2e/kubectl/kubectl.go:1845]: Adding a docstring explaining this function would be good.

Comment (dougsland) [inline test/e2e/kubectl/kubectl.go:1845]: Why findOcc ? I think in golang we don't have limitations in chars for func names. I would go with a full name for easy understanding. findOccurrences()? Today we understand what's occ in 3 months ahead would be difficult without digging the code.

File: test/e2e/kubectl/kubectl.go
Comment (dougsland) [inline test/e2e/kubectl/kubectl.go:1855]: maybe docstring here too?

**Review by dougsland — COMMENTED** [review]
it seem sane patch, just few comments.

File: test/e2e/kubectl/kubectl.go
Comment (SaiHarshaK) [reply test/e2e/kubectl/kubectl.go:1855]: I thought we add comments only to exported functions, will add

**Comment by dougsland** [comment]
LGTM assuming happy checks.

File: test/e2e/kubectl/kubectl.go
Comment (soltysh) [inline test/e2e/kubectl/kubectl.go:1513]: I was actually expecting something simpler, along the lines of:
```go
requiredItems := []string{"Client Version: ", "Server Version: "}
for _, item := range requiredItems {
	if matched, _ := regexp.MatchString(item+`version.Info\{Major:"\d", Minor:"[\d\+]+", GitVersion:"v\d\.\d+.[\w\-\.\+]+", GitCommit:"[0-9a-f]+"`, output); !matched {
		fmt.Printf("didn't match %s\n", item)
	}
}
```

You could tweak the regexp and/or items, but that should be sufficient.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: test/e2e/kubectl/kubectl.go
Comment (SaiHarshaK) [reply test/e2e/kubectl/kubectl.go:1513]: Will update

Comment (SaiHarshaK) [reply test/e2e/kubectl/kubectl.go]: Previously, I wasn't sure if the order of items Major, Minor, GitVersion,... will be in same order always, so i split the string to check. I have it updated now.

File: test/e2e/kubectl/kubectl.go
Comment (soltysh) [inline test/e2e/kubectl/kubectl.go:1511]: This should not be needed.

**Review by soltysh — APPROVED** [review]
/priority backlog
/triage accepted
/lgtm
/approve

File: test/e2e/kubectl/kubectl.go
Comment (SaiHarshaK) [reply test/e2e/kubectl/kubectl.go]: I have updated it but LGTM label was removed, could you add it again?

**Pull request merged**
